cabal-install/Distribution/Client/FileMonitor.hs

```diff
@@ -199,7 +199,7 @@ monitorFileHashedSearchPath notFoundAtPaths foundAtPath =
 -- files to be monitored (index by their path), and a list of
 -- globs, which monitor may files at once.
 data MonitorStateFileSet
-   = MonitorStateFileSet !(Map FilePath MonitorStateFile)
```

Comment:
And all the other changes in that patch follow more or less mechanically from that change of representation.